packages/shopping-cart/src/shopping-cart-reducer.ts

```diff
@@ -85,29 +66,47 @@ function shouldQueueReducerEvent( cacheStatus: CacheStatus, action: ShoppingCart
 	return false;
 }
 
-function shoppingCartReducer(
+export function reducerWithQueue(
 	state: ShoppingCartState,
 	action: ShoppingCartAction
 ): ShoppingCartState {
-	const couponStatus = state.couponStatus;
+	if (
+		cacheStatusesForIgnoringReload.includes( state.cacheStatus ) &&
+		action.type === 'CART_RELOAD'
+	) {
+		debug( 'cart is pending an operation; ignoring reload action' );
+		return state;
+	}
 
 	// If the cacheStatus is 'fresh' or 'pending', then the initial cart has not
 	// yet loaded and so we cannot make changes to it yet. We therefore will
 	// queue any action that comes through during that time except for
 	// 'RECEIVE_INITIAL_RESPONSE_CART' or 'RAISE_ERROR'.
 	if ( shouldQueueReducerEvent( state.cacheStatus, action ) ) {
-		if ( action.type === 'CART_RELOAD' ) {
-			debug( 'cart has not yet loaded; ignoring reload action', action );
-			return state;
-		}
 		debug( 'cart has not yet loaded; queuing requested action', action );
 		return {
 			...state,
 			queuedActions: [ ...state.queuedActions, action ],
 		};
 	}
 
+	state = shoppingCartReducer( state, action );
+
+	if ( shouldPlayQueuedActions( state ) ) {
+		const dispatch = ( queuedAction: ShoppingCartAction ) =>
+			( state = shoppingCartReducer( state, queuedAction ) );
+		playQueuedActions( state, dispatch );
+	}
+
+	return state;
```

Comment:
The `playQueuedAction` code can be written as:
```js
if ( shouldPlay( state ) ) {
  const actions = [ ...state.queuedActions, { type: 'CLEAR_QUEUED_ACTIONS' } ];
  state = actions.reduce( shoppingCartReducer, state );
}
```

Reply:
Haha, oh yeah, it's a _reducer_ 🤦